homeassistant/components/fibaro/sensor.py

```diff
@@ -72,44 +76,31 @@ def __init__(self, fibaro_device):
         super().__init__(fibaro_device)
         self.entity_id = f"{DOMAIN}.{self.ha_id}"
         if fibaro_device.type in SENSOR_TYPES:
-            self._unit = SENSOR_TYPES[fibaro_device.type][1]
-            self._icon = SENSOR_TYPES[fibaro_device.type][2]
-            self._device_class = SENSOR_TYPES[fibaro_device.type][3]
+            self.entity_description = SENSOR_TYPES[fibaro_device.type]
+            self._attr_native_unit_of_measurement = (
+                self.entity_description.native_unit_of_measurement
+            )
         else:
-            self._unit = None
-            self._icon = None
-            self._device_class = None
+            self.entity_description = SENSOR_TYPES["unknown"]
+            self._attr_native_unit_of_measurement = None
         with suppress(KeyError, ValueError):
-            if not self._unit:
+            if not self._attr_native_unit_of_measurement:
```

Comment:
If you take the above suggestions, this statement would change to look at `self.entity_description.native_unit_of_measurement`